src/combat.ts

```diff
@@ -615,46 +619,90 @@ export class Macro extends StrictMacro {
   }
 }
 
+function customizeMacro<M extends StrictMacro>(macro: M) {
+  if (have($effect`Eldritch Attunement`)) {
+    return Macro.if_($monster`Eldritch Tentacle`, Macro.basicCombat()).step(macro);
+  } else if (getTodaysHolidayWanderers().length !== 0) {
+    return Macro.ifHolidayWanderer(
+      Macro.externalIf(
+        haveEquipped($item`backup camera`) &&
+          get("_backUpUses") < 11 &&
+          get("lastCopyableMonster") === $monster`Knob Goblin Embezzler` &&
+          myFamiliar() === meatFamiliar(),
+        Macro.skill($skill`Back-Up to your Last Enemy`).step(Macro.load()),
+        Macro.basicCombat()
+      )
+    ).step(macro);
+  } else {
+    return macro;
+  }
+}
+
+function makeCcs<M extends StrictMacro>(macro: M) {
+  writeCcs(`[default]\n"${customizeMacro(macro).toString()}"`, "garbo");
+  cliExecute("ccs garbo");
+}
+
+function completeCombat<T>(action: () => T) {
+  try {
+    const result = action();
+    while (inMultiFight()) runCombat();
+    if (choiceFollowsFight()) visitUrl("choice.php");
+    return result;
+  } catch (e) {
+    throw `Combat exception! Last macro error: ${get("lastMacroError")}. Exception ${e}.`;
+  }
+}
```

Comment:
IMO it is not clear from the name that `action` should be an action that initiates combat. To me, `completeCombat` sounds like a function that finishes an already-started combat, and `action` sounds like something that happens _during_ combat.

Reply:
renamed